### PR TITLE
manager-io: init 24.12.23.1999

### LINKED ISSUE #367590: Package request: managerio
## Project description

A free and fully featured accounting software for small businesses and individuals.

## Metadata

* homepage URL: https://www.manager.io/
* source URL: https://github.com/Manager-io/Manager/releases/
* license: (they have not given)
* platforms: linux, macos, windows

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by darwincereska**
Closes #367590 adds manager from Manager-io

Accounting software. Available for Windows, Mac and Linux.
Homepage: [manager.io](https://manager.io)

Notes: Package compiles but you need libicu 
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ma/managerio/package.nix [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+appimageTools.wrapType2 rec {
+  pname = "manager";
+  version = "24.12.23.1999";
+  src = fetchurl {
+    url = "https://github.com/Manager-io/Manager/releases/download/${version}/Manager-linux-x64.AppImage";
+    hash = "sha256-Q4bH1cFvZfNSOWGXmg/RAOtjK6u5p2iRfUrOvetnoOs=";
+  };
+  meta = {
+    description = "Accounting software. Available for Windows, Mac and Linux.";
+    homepage = "https://github.com/Manager-io/Manager";
```
Comment (pluiedev) [inline pkgs/by-name/ma/managerio/package.nix:15]: I think this is a better homepage
```suggestion
    homepage = "https://www.manager.io/";
```

File: pkgs/by-name/ma/managerio/package.nix [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+}:
+appimageTools.wrapType2 rec {
+  pname = "manager";
+  version = "24.12.23.1999";
+  src = fetchurl {
+    url = "https://github.com/Manager-io/Manager/releases/download/${version}/Manager-linux-x64.AppImage";
+    hash = "sha256-Q4bH1cFvZfNSOWGXmg/RAOtjK6u5p2iRfUrOvetnoOs=";
+  };
+  meta = {
+    description = "Accounting software. Available for Windows, Mac and Linux.";
```
Comment (pluiedev) [inline pkgs/by-name/ma/managerio/package.nix:14]: Descriptions should not be complete sentences but fragments (they should complete a sentence like "<package name> is ...")
```suggestion
    description = "Free accounting software for Windows, Mac and Linux";
```

File: pkgs/by-name/ma/managerio/package.nix [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    homepage = "https://github.com/Manager-io/Manager";
+    maintainers = with lib.maintainers; [ darwincereska ];
```
Comment (pluiedev) [inline pkgs/by-name/ma/managerio/package.nix:16]: Need at least `meta.platforms`, `meta.license` (unknown - presumably freeware...?), `meta.sourceProvenance`
```suggestion
    maintainers = with lib.maintainers; [ darwincereska ];
    platforms = [
      "x86_64-linux"
      "x86_64-darwin"
      "aarch64-darwin"
      # "x86_64-windows" # Windows ones are optional, really, feel free to exclude them
      # "i686-windows"
   ];
   license = with lib.licenses; [ free ]; # Unspecified freeware
   sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
```

File: pkgs/by-name/ma/managerio/package.nix [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  appimageTools,
+  fetchurl,
+}:
+appimageTools.wrapType2 rec {
+  pname = "manager";
+  version = "24.12.23.1999";
+  src = fetchurl {
+    url = "https://github.com/Manager-io/Manager/releases/download/${version}/Manager-linux-x64.AppImage";
+    hash = "sha256-Q4bH1cFvZfNSOWGXmg/RAOtjK6u5p2iRfUrOvetnoOs=";
+  };
```
Comment (pluiedev) [inline pkgs/by-name/ma/managerio/package.nix:12]: Would be best to add Darwin support too (via `undmg`)

File: pkgs/by-name/ma/managerio/package.nix (hunk 3 above)
Comment (pluiedev) [inline pkgs/by-name/ma/managerio/package.nix:16]: Need at least `meta.platforms`, `meta.license` (unknown - presumably freeware...?), `meta.sourceProvenance`
```suggestion
    maintainers = with lib.maintainers; [ darwincereska ];
    platforms = [
      "x86_64-linux"
      "x86_64-darwin"
      "aarch64-darwin"
      # "x86_64-windows" # Windows ones are optional, really, feel free to exclude them
      # "i686-windows"
   ];
   license = with lib.licenses; [ free ]; # Unspecified freeware
   sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
```

**Review by pluiedev — CHANGES_REQUESTED** [review]

**Comment by darwincereska** [comment]
> Please compress the submission record. There should be two submission records
> ```
> maintainers: add darwincereska
> ```
> ```
> manager-io: init at 24.12.23.1999
> ```
> Modify the pr title to
> 
> ```
> manager-io: init at 24.12.23.1999
> ```

The file gets errors if I'm not in the maintainer list

**Pull request closed without merging**